### DISCUSSION TITLE
Responsive objects in the background

### DISCUSSION THREAD

**Discussion by niawjunior** [fabricjs/fabric.js#9545] [Q&A] [answered]
Hi, I have a question about responsive objects in the background. How can we adjust the size of these objects based on the background size? For instance, if I start drawing something on a mobile screen and then switch to desktop size, how can we ensure that the objects' sizes adapt while maintaining their positions on the background?

[mobile]
<img width="1440" alt="image" src="https://github.com/fabricjs/fabric.js/assets/17493195/29863275-2f40-4a90-b6cf-acf7e1ef29a9">

[desktop]
<img width="1440" alt="image" src="https://github.com/fabricjs/fabric.js/assets/17493195/b906eeae-ae5d-45a7-a709-f1b9f72bb631">

**Comment by ShaMan123** [comment, marked as answer]
Did you try to reposition the objects when the canvas resizes?
Calculate a % based distance and position using that.
You can hook into the lifecycle or listen to a ResizeObserver, the first seems better to me

https://github.com/fabricjs/fabric.js/blob/e4ac8b5284b2c1be80b6174d9b8d31e41136e8ce/src/canvas/StaticCanvas.ts#L321-L341

**Reply by niawjunior** [reply]
Hi thanks for the answer
 
Do you have any demo links for this?

this is my resize canvas function

```
  @HostListener('window:resize', ['$event'])
  resizeCanvas(event?: any) {
    if (this.canvasFabric) {
      const width = window.innerWidth > 0 ? window.innerWidth : screen.width;
      const height =
        window.innerHeight > 0 ? window.innerHeight : screen.height;
      const aspect = this.imgFabric.width! / this.imgFabric.height!;
      const imageWidth = width;
      const imageHeight = width / aspect;
      const canvasHeight = Math.max(height, imageHeight);
      this.canvasFabric.setWidth(imageWidth);
      this.canvasFabric.setHeight(canvasHeight);
      var topPosition = (canvasHeight - imageHeight) / 2;

      this.imgFabric.set({
        top: topPosition,
        scaleX: imageWidth / this.imgFabric.width!,
        scaleY: imageHeight / this.imgFabric.height!,
      });
    }
  }
```

https://github.com/fabricjs/fabric.js/assets/17493195/d8a7cc0b-c3f5-41e6-a218-898709c4b00f

**Reply by ShaMan123** [reply]
provide what you have so I can look into it

**Reply by niawjunior** [reply]
Hi this one https://stackblitz.com/edit/js-nsu2bn?file=index.js

**Reply by niawjunior** [reply]
My expectation every object still at the same position and can work any screen

https://github.com/fabricjs/fabric.js/assets/17493195/b61c16e8-4bd4-4ff6-9e1a-801063dd1462

**Reply by ShaMan123** [reply]
I work with v6 repros only, sorry
